### ISSUE TITLE
Deep cloning options is breaking stylus node objects

### ISSUE THREAD

**Issue by Bockit** [stevelacy/gulp-stylus#101]
In https://github.com/stevelacy/gulp-stylus/commit/15fb57a81bfcbf927a84c14e364a2889128997d8 a change was made to deep clone the options object. This breaks any stylus nodes you've passed in, e.g., 

```javascript
.pipe(stylus({ define: {
    'text-colour': new stylusRaw.nodes.RGBA(255, 0, 0, 1)
}}))
```

Deep cloning turns the stylus RGBA node into just an object with properties. Because nodes have a reference to themselves, when stylus tries to coerce the object into a stylus object node it ends up recursing on the self-reference until you get a stack overflow.

**Comment by Bockit** [comment]
Is it possible for this change to be published?

**Comment by stevelacy** [comment]
I was planning to merge this with the sourcemap support and publish as one. 
I'll publish

**Comment by stevelacy** [comment]
Published with sourcemaps @1.3.6
